client/src/containers/App.tsx

```diff
@@ -42,13 +48,169 @@ export default class App extends Component<Props, State> {
 
     this.state = {
       showEngageModal: false,
+      entryIndex: 0,
+      entries: [
+        {
+          index: "1 of 4",
+          title: "Timeline Tab",
+          description: "Click the Timeline tab to view info about your building over time.",
+          img: Timeline_gif,
+        },
+        {
+          index: "2 of 4",
+          title: "Spanish Support",
+          description:
+            "Who Owns What is now available in Spanish. Click “ES” in the upper right corner to switch your language.",
+          img: Spanish_gif,
+        },
+        {
+          index: "3 of 4",
+          title: "Unique Tab URLs",
+          description:
+            "It's now possible to share links to specific tabs (Overview, Timeline, Portfolio, & Summary).",
+          img: URLS_gif,
+        },
+        {
+          index: "4 of 4",
+          title: "Last Sold",
+          description: "The Overview and Portfolio tabs now display the date and price from when your building was last sold.",
+          img: LastSold_gif,
+        },
+      ],
     };
   }
 
+  nextEntry() {
+    this.state.entryIndex === this.state.entries.length - 1
+      ? this.setState({ entryIndex: 0 })
+      : this.setState({ entryIndex: this.state.entryIndex + 1 });
+  }
+
+  prevEntry() {
+    this.state.entryIndex === 0
+      ? this.setState({ entryIndex: this.state.entries.length - 1 })
+      : this.setState({ entryIndex: this.state.entryIndex - 1 });
+  }
+
+  toggleWidget(widget: any) {
+    let tooltipTriangle = document.querySelector(".widget-tooltip-triangle");
+    let infoButton = document.querySelector(".widget-button-info");
+    if (widget.style.display === "none") {
+      widget.style.display = "inline-block";
+      if (tooltipTriangle) tooltipTriangle.classList.add("toggled");
+      if (infoButton) infoButton.classList.add("pressed");
+    } else {
+      widget.style.display = "none";
+      if (tooltipTriangle) tooltipTriangle.classList.remove("toggled");
+      if (infoButton) infoButton.classList.remove("pressed");
+    }
```

Comment:
Hmm, so this code will be the easiest to understand and modify if you do things 100% using DOM APIs or 100% React, rather than mixing the two.  Right now this seems to be mixing the two, in that it's creating the widget using React but then modifying it using DOM APIs, which will end up confusing React because it's [virtual DOM](https://reactjs.org/docs/faq-internals.html) now deviates from the real DOM.  Instead of that, you'll likely want to do things 100% the React way by modifying the component's state when something is clicked, and then changing the `render()` method to apply the appropriate classes and styling depending on the state.  The [classnames](https://www.npmjs.com/package/classnames) package can be particularly useful in generating the final list of classes, I'm not sure if we use it yet in this project or not.

(I'm realizing now that we talked about doing parts of the first version of the widget using DOM APIs because of how hard it is to "point at" elements in completely different parts of a react app, but now that we're not doing the "pointing at" thing and just embedding animated GIFs, I don't think that mixing the two styles is very helpful... sorry if that's confusing, happy to talk it over more!)

Reply:
I attempted this, please lmk if it actually is React-only now